compiler/src/iree/compiler/Dialect/LinalgExt/Transforms/TilingInterfaceImpl.cpp

```diff
@@ -1704,45 +1704,53 @@ AttentionOp::getTiledImplementation(OpBuilder &builder,
 
   Location loc = getLoc();
 
-  SmallVector<Range> querySlice =
-      getPermutedSlice(getQueryMap(), offsets, sizes);
-  SmallVector<Range> keySlice = getPermutedSlice(getKeyMap(), offsets, sizes);
-  SmallVector<Range> valueSlice =
-      getPermutedSlice(getValueMap(), offsets, sizes);
-  SmallVector<Range> outputSlice =
-      getPermutedSlice(getOutputMap(), offsets, sizes);
-
-  Value scale = getScale();
-
   SmallVector<Value> tiledOperands;
+
+  SmallVector<Range> querySlice = getPermutedSlice(getQueryMap(), offsets, sizes);
   tiledOperands.emplace_back(getSlice(builder, loc, getQuery(), querySlice));
-  tiledOperands.emplace_back(getSlice(builder, loc, getKey(), keySlice));
+
+  SmallVector<Range> keySlice = getPermutedSlice(getKeyMap(), offsets, sizes);
+   tiledOperands.emplace_back(getSlice(builder, loc, getKey(), keySlice));
+
+  SmallVector<Range> valueSlice = getPermutedSlice(getValueMap(), offsets, sizes);
   tiledOperands.emplace_back(getSlice(builder, loc, getValue(), valueSlice));
+
+  Value scale = getScale();
   tiledOperands.emplace_back(scale);
+
+  int hasMask = 0;
+  std::optional<Value> attnMask = getMask();
+  if (getMask()) {
+    SmallVector<Range> maskSlice = getPermutedSlice(*getMaskMap(), offsets, sizes);
+    tiledOperands.emplace_back(getSlice(builder, loc, attnMask.value(), maskSlice));    
+    hasMask++;
+  } else {
+    tiledOperands.emplace_back(Value());
+  }
+
+  SmallVector<Range> outputSlice = getPermutedSlice(getOutputMap(), offsets, sizes);
   tiledOperands.emplace_back(getSlice(builder, loc, getOutput(), outputSlice));
 
   std::optional<Value> max = getMax();
-  if (max) {
-    SmallVector<Range> maxSlice =
-        getPermutedSlice(*getMaxMap(), offsets, sizes);
+  if (std::optional<Value> max = getMax()) {
```

Comment:
NIT: Is there a reason we refactored this? seems functionally the same as pre-change

Reply:
No reason, reverted